SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -762,15 +762,55 @@ public void SelectAllocationsByTeamAllocationStatus()
             _classUnderTest.CreateAllocation(teamAllocationRequest);
             _classUnderTest.CreateAllocation(teamAndWorkerAllocationRequest);
 
-            var (workerAllocations, _) = _classUnderTest.SelectAllocations(0, 0, null, workerTeam.TeamId, teamAllocationStatus: "allocated");
+            var (workerAllocations, _) = _classUnderTest.SelectAllocations(0, 0, null, workerTeam.TeamId, 0, teamAllocationStatus: "allocated");
 
             workerAllocations.Count.Should().Be(1);
             workerAllocations.FirstOrDefault().AllocatedWorker.Should().NotBeNull();
 
         }
 
         [Test]
+        public void GetAllocationsByAllocationIdReturnsOnlyAllocationWithProvidedId()
+        {
+            // Create worker and teams
+            var worker = TestHelpers.CreateWorker(hasAllocations: false, hasWorkerTeams: false, id: 123);
+            var anotherWorker = TestHelpers.CreateWorker(hasAllocations: false, hasWorkerTeams: false, id: 124);
+
+            var workerTeam = TestHelpers.CreateWorkerTeam(worker.Id);
+            var anotherWorkerTeam = TestHelpers.CreateWorkerTeam(anotherWorker.Id);
+
+            worker.WorkerTeams = new List<WorkerTeam> { workerTeam };
+            anotherWorker.WorkerTeams = new List<WorkerTeam> { anotherWorkerTeam };
+
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.WorkerTeams.Add(workerTeam);
+            DatabaseContext.Workers.Add(anotherWorker);
+            DatabaseContext.WorkerTeams.Add(anotherWorkerTeam);
+            DatabaseContext.SaveChanges();
+
+            var (createAllocationRequest, _, allocator, resident, _) = TestHelpers.CreateAllocationRequest(workerId: worker.Id, teamId: workerTeam.TeamId);
+            var (createAnotherAllocationRequest, _, anotherAllocator, anotherResident, _) = TestHelpers.CreateAllocationRequest(workerId: anotherWorker.Id, teamId: anotherWorkerTeam.TeamId);
+
+            DatabaseContext.Workers.Add(allocator);
+            DatabaseContext.Persons.Add(resident);
+            DatabaseContext.Workers.Add(anotherAllocator);
+            DatabaseContext.Persons.Add(anotherResident);
+            DatabaseContext.SaveChanges();
+
+            _classUnderTest.CreateAllocation(createAllocationRequest);
```

Comment:
Would it be better to add the allocations directly to the DatabaseContext, so the test is not relying on other piece of functionality?